libdevcore/termcolor.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libdevcore/termcolor.h>
+
+#include <cassert>
+#include <cstdio>
+#include <iostream>
+#include <ostream>
+#include <utility>
+
+#if defined(__unix__)
+#include <unistd.h>
+#endif
+
+#if defined(_WIN32)
+#include <windows.h>
+#endif
+
+using namespace std;
+
+namespace termcolor {
+
+namespace internal {
+
+static int const enabled_index = std::ios_base::xalloc();
+
+bool stdoutInitialized = initialize(STDERR_FILENO);
+bool stderrInitialized = initialize(STDERR_FILENO);
+
+}  // namespace internal
+
+bool initialize(int terminalFd)
+{
+	if (terminalFd != STDOUT_FILENO && terminalFd != STDERR_FILENO)
+		return false;
+
+#if defined(_WIN32) && defined(ENABLE_VIRTUAL_TERMINAL_PROCESSING)
+	static const int fdMap[] = { STD_INPUT_HANDLE, STD_OUTPUT_HANDLE, STD_ERROR_HANDLE };
+
+	// Set output mode to handle virtual terminal (ANSI escape sequences)
+	// ignore any error, as this is just a "nice-to-have"
+	// only windows needs to be taken care of, as other platforms (Linux/OSX) support them natively.
+	HANDLE hOut = GetStdHandle(fdMap[terminalFd]);
```

Comment:
This assumes that `STDOUT_FILENO` is at most 2. You could use an `std::array` and `.at()`.